presto-hive/src/main/java/io/prestosql/plugin/hive/HivePartitionManager.java

```diff
@@ -223,7 +226,9 @@ public HiveTableHandle applyPartitionResult(HiveTableHandle handle, HivePartitio
                 partitions.getBucketHandle(),
                 partitions.getBucketFilter(),
                 handle.getAnalyzePartitionValues(),
-                handle.getAnalyzeColumnNames());
+                handle.getAnalyzeColumnNames(),
+                Optionals.combine(handle.getConstraintColumns(), columns,
+                        (oldColumns, newColumns) -> Sets.union(oldColumns, newColumns)));
```

Comment:
Combine old and new column references here. I copied io.prestosql.util.Optionals into io.prestosql.plugin.hive.util as I didn't see an obvious module to move it to.